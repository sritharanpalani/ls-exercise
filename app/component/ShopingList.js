import React from 'react';
import {
  View,
  StyleSheet,
  Text,
  Dimensions,
  FlatList,
  Image,
} from 'react-native';
const width = Dimensions.get('window').width;

const styles = StyleSheet.create({
  container: {
    flex: 2,
    width: width,
    marginTop: 8,
  },
  headerContainer: {
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
  },
  itemContainer: {
    display: 'flex',
    flexDirection: 'row',
    margin: 16,
  },
  imageContainer: {
    display: 'flex',
  },
  image: {
    display: 'flex',
    width: 200,
    height: 200,
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    color: '#000000',
    textAlign: 'left',
  },
  detailsContainer: {
    display: 'flex',
    justifyContent: 'flex-start',
    marginTop: 16,
  },
  currencyContainer: {
    display: 'flex',
    flexDirection: 'row',
  },
  currency: {
    fontWeight: 'bold',
    marginRight: 16,
  },
});

const ShopingList = props => {
  const {title, products} = props;

  const keyExtractor = item => `${item.name} ${item.price} ${item.key}`;
  const renderItem = ({item}) => (
    <View style={styles.itemContainer}>
      <View style={styles.imageContainer}>
        <Image source={{uri: item.url}} style={styles.image} />
        <View style={styles.detailsContainer}>
          <Text>{item.name}</Text>
          <View style={styles.currencyContainer}>
            <Text style={styles.currency}>{item.currency}</Text>
            <Text>{parseFloat(item.price).toFixed(2)}</Text>
          </View>
        </View>
      </View>
    </View>
  );
  return (
    <View style={styles.container}>
      <View style={styles.headerContainer}>
        <Text style={styles.title}>{title}</Text>
      </View>
      <View>
        <FlatList
          data={products}
          keyExtractor={keyExtractor}
          renderItem={renderItem}
          horizontal
          extraData={props}
        />
      </View>
    </View>
  );
};

export default ShopingList;
