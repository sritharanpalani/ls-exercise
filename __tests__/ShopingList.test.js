import * as React from 'react';
import renderer from 'react-test-renderer';

import ShopingList from '../app/component/ShopingList';

test('<ShopingList> renders correctly', () => {
  const tree = renderer.create(<ShopingList />).toJSON();
  expect(tree).toMatchSnapshot();
});
