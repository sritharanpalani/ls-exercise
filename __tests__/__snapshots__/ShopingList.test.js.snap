// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShopingList> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 2,
      "marginTop": 8,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "left",
        }
      }
    />
  </View>
  <View>
    <RCTScrollView
      disableVirtualization={false}
      extraData={Object {}}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
